promql/parse_test.go

```diff
@@ -779,6 +786,10 @@ var testExpr = []struct {
 		input:  "foo == on(bar) 10",
 		fail:   true,
 		errMsg: "vector matching only allowed between instant vectors",
+	}, {
+		input:  "foo + group_left(baz) bar",
```

Comment:
Why is this invalid? Looks fine to me.

Reply:
Because `group_left|group_right` are only allowed together with `on|ignoring` by the old parser. This test makes sure that the new parser behaves the same.

http://demo.robustperception.io:9090/graph?g0.range_input=1h&g0.expr=foo%20%2B%20group_left(baz)%20bar&g0.tab=1

Reply:
Ah yes, it'd be valid but useless.